### DISCUSSION TITLE
Github webhooks triggers a multibranch scan of all refs

### DISCUSSION THREAD

**Discussion by zagadishreddy** [community/community#58041] [Enterprise]
### Select Topic Area

General

### Body

Hi Team, 

Currently we migrated to GIT HUB Enterprise version, and we are using the jenkins as CI Tools and configured Multi Branch Pipeline.
We are using Moo Repo and configured as Multibranch pipeline  . In that Mono Repo we have 100 SubRepos we have . 
 
. We have a large repository - and so many commits every day. Every time a webhook is fired from github and hits jenkins all N multibranch pipelines begin scanning ALL remote refs for the repository. 

can you please help us instead of scanning all sub repos  , need to scan only specific sub repo only. Kindly suggest on this one .

Thanks
Jagadish Reddy

**Comment by anhtran917** [comment]
Hi Jagadish Reddy,

I think you would like to scan only a specific sub-repo instead of scanning all sub-repos.

To achieve this, you can use the "Filter by name" option in the "Branch Sources" section of your Jenkinsfile. This option allows you to specify a regular expression to filter the branches that you want to build.

For example, if you want to scan only the sub-repo named "sub-repo-1", you can use the following regular expression in the "Filter by name" option:

`sub-repo-1.*`
This regular expression will match all branches that start with "sub-repo-1".

You can also use other regular expressions to match different patterns of branch names.

I hope this helps. Let me know if you have more questions.

Thanks, 
Anh

**Comment by tuves** [comment]
Hi @zagadishreddy,

The GitHub Community on Discussions is a space for all GitHub users to come and discuss ideas and projects. Your post was removed because it contained content that violates our Community Code of Conduct, including posting the same comment in multiple places. As a result, we have hidden the duplicate comments, and we ask that you no longer cross-post.

For additional guidance, please review our [Community on Discussions Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md) and [GitHub’s Terms of Service](https://docs.github.com/en/github/site-policy/github-terms-of-service).
